### ISSUE TITLE
Android Material redesign

### ISSUE THREAD

**Issue by MCMrARM** [MCMrARM/revolution-irc#172]
Android P has pretty much started a notable Material redesign. While, I do not like the new changes, the application still probably should be redesigned in order to be more consistent with the system.

I'd like to hear your opinion on this in this issue.

**Comment by MCMrARM** [comment]
I don't think so.

**Comment by MCMrARM** [comment]
New UI concept:
![Home](https://mrarm.io/u/Clipboard%202019-02-14%2022-17-01.png)
![Channel](https://mrarm.io/u/Clipboard%202019-02-14%2022-17-44.png)
Note that the old UI is not going away anywhere, and this is to be considered a second option (most likely default though);.
The main reason I want to redesign the UI is so we can use slide-from-left to exit to the channel list screen, which allows in more flexibility for grouping.
I'd like to ask for your feedback.

**Comment by MCMrARM** [comment]
Concept 2:
![https://mrarm.io/u/Clipboard%202019-02-15%2022-19-36.png](https://mrarm.io/u/Clipboard%202019-02-15%2022-19-36.png)

**Comment by MCMrARM** [comment]
Padding fix:
![Concept 1](https://mrarm.io/u/Clipboard%202019-02-15%2022-31-40.png)
![Concept 2](https://mrarm.io/u/Clipboard%202019-02-15%2022-31-58.png)

**Comment by MCMrARM** [comment]
Here's how the current new-ui branch looks like (video, still uses legacy theme): http://mrarm.io/u/device-2019-02-21-190654.mp4

**Comment by target111** [comment]
Looks good but maybe you should group channels under the server button so that one doesn't have to scroll too much if they have lots of server with lots of channels and the one they wanna reach is at the bottom of the list

**Comment by MCMrARM** [comment]
I don't understand - what do you want to group them by?

**Comment by MCMrARM** [comment]
Overall there's going to be a Search button and maybe a scrollbar as well in the Servers tab, I think.

**Comment by target111** [comment]
From what I was able to see in the video, the server window probably displays all network names along with the channels that you've joined
on that network, underneath. If one would want to reach a channel from a network that is at the bottom off the list, they will have to scroll through a lot of stuff in case there are many networks present above, so what I meant is: in the server window, display only network names and when those names are clicked, list the channels currently opened on that network. If none of what I say makes sense then it's probably my fault because I couldn't understand the interface from that short video very well.

**Comment by kairusds** [comment]
you should also add a vertical scrollview for the channels list instead of it being horizontal to make it easier to find what channel you're looking for and a user list similar to discord without the profile images with different groups

**Comment by kairusds** [comment]
fullscreen option is also much appreciated

**Comment by MCMrARM** [comment]
A good question is how the chats should actually be organized:
1) The all-chats-go-to-one-place view
 - How customizable should it be?
    - Should you be able to disable displaying last message for all channels? Probably yes as that'd make it more compact.
 - We probably want different ways to sort it
   - Alphabetical sort needs a letter scrollbar
   - Does a recent-based chat even make sense? IRC is not exactly like normal chat messaging apps.
   - Are there any other approaches we could consider? Perhaps some sort of algorithm could be used to decide the order or even a neural network?
2) Server-grouping based
 - When would it be preferable over the chat list?
 - Do the server groups need to be collapsible?
 - Is a quick way to disconnect required?
3) Are there any other viable ways to group the channels?

**Comment by MCMrARM** [comment]
I'd appreciate your thoughts on the last comment, as I am unable to answer the questions in a way that can satisfy everyone myself, and there'll probably be no going back from the new UI.

**Comment by benrob0329** [comment]
>Does a recent-based chat even make sense? 

Yes, people (including me) may want to know what channel was active last.

>Are there any other approaches we could consider? Perhaps some sort of algorithm could be used to decide the order or even a neural network?

Perhaps most active overall?

>When would it be preferable over the chat list?

Most of the time if someone has quite a few channels. One of the things I've seen a lot of people choose Matrix or Discord over IRC for is a lack of organization, which could present other features for grouping channels but I think that the simplest form is per-server grouping.

>Do the server groups need to be collapsible?

Probably not, this will add another button onto the list and will likely be annoying while scrolling if accidentally pressed.

>Is a quick way to disconnect required?

Long-pressing on the channel or server for a context menu would be enough. This menu should also have a checkbox for auto join/connect.

>Are there any other viable ways to group the channels?

By prefix may work, for example there are several `#minetest-` channels, several `#debian-` channels, and several `#kde-` channels that could all be grouped together. This would need to ignore extra #s however due to Freenode's double-hash prefix for offtopic channels.

**Comment by MCMrARM** [comment]
Also, additional questions for channel grouping:

Feature description: Channel grouping would let you group together channels that you frequently access. Usage examples: Important Chats (group: #revolutionirc, #privatechat1, #privatechat2, anotheruser); Work (#general - Slack 1 IRC Bridge, #support - IRC Bridge 2).
Ways the channel groups could be used for:
- Show tabs for the channels in the group in the new UI.
- Show the channel groups in the channel list.

Questions:
- Would you use such a feature?
- What groups would you create?
- Do you think the same groups would make sense to be both used for the in-chat UI tabs and groups in the channel list?

**Comment by MCMrARM** [comment]
The channel groups would additionally replace the current server tab system, which is going away with the new UI.

**Comment by benrob0329** [comment]
Quick UI concept for displaying groups:

![IMG_20190402_143004](https://user-images.githubusercontent.com/19180682/55426921-f1ddc400-5553-11e9-8ef8-a5dd7d2555b4.jpg)

**Comment by MCMrARM** [comment]
Re-redesign design:
![Main screen](https://mrarm.io/u/Paste%202019-04-03%2013-14-36.svg)

Customizable server icons at top, server groups in the main view, view types on bottom.

**Comment by target111** [comment]
Is the redesign arriving to the play store  any time soon?

**Comment by MCMrARM** [comment]
Because of the advancement of the Android UI framework, the way I built the new UI feels weird and would make further maintenance more difficult. As such I consider the newui branch dead and if the new UI is to happen it should be rewritten from scratch. Some of the ideas that were part of the new UI should also be reconsidered.

I will try to find some time soon to rewrite the chat view in the current code as well as the other recyclerview adapters to a more modern pattern (we have ConcatAdapter now in the SDK). I was writing an Android app in Kotlin recently and was able to use these new UI components.
